src/packages/Controls/Drawing/Drawing.js

```diff
@@ -1938,9 +1938,13 @@ var Drawing = class Drawing extends Control {
      * and toggles event 'mousemove' on map.
      *
      * @method onShowDrawingClick
+     * @param { event } e évènement associé au clic
      * @private
      */
-    onShowDrawingClick () {
+    onShowDrawingClick (e) {
+        if (e.target.ariaPressed === "true") {
```

Comment:
Pourquoi ne pas utiliser la variable opened définie juste en dessous qui semble faire le taf ?

Applicable pour tous les autres widgets a priori ?

Reply:
la variable opened est défini dans chaque widget avec le nom spécifique du widget 
var opened = this._showLayerSwitcherButton.ariaPressed;

Comme on a pas de classe générique pour gérer les widget il faut copier coller le code dans chaque widget. 
l'avantage de passer par l'évènement est que je copie colle une fois (avec seulement quelques modifs spécifique à chaque widget)
sinon je dois chercher et écrire dans chaque widget ce fameux showWidgetNameButton sachant que ce nom n'est pas standardisé.

Pour plus de clarté / factorisation dans le code il faut s'attaquer au sujet des classes génériques dans les extensions